projects/RabbitMQ.Client/client/impl/ConsumerDispatching/ConsumerDispatcherChannelBase.cs

```diff
@@ -120,23 +119,23 @@ public WorkStruct(WorkType type, IBasicConsumer consumer, string consumerTag)
                 : this()
             {
                 WorkType = type;
-                Consumer = consumer;
+                Strategy = DispatchStrategyFactory.Create(consumer);
```

Comment:
This creates one Strategy per item processed, resulting in a negative impact in performance.